src/main/java/com/productService/exception/ProductServiceExceptionHandler.java

```diff
@@ -10,8 +12,14 @@
 public class ProductServiceExceptionHandler {
 
     @ExceptionHandler(value = NotFoundException.class)
-    public ResponseEntity<String> notFoundException(NotFoundException exception) {
+    public ResponseEntity<ExceptionResponse> notFoundException(NotFoundException exception) {
         log.info("Entering ProductServiceExceptionHandler.notFoundException with parameter exception {}.", exception);
-        return ResponseEntity.badRequest().body(exception.toString());
+        return ResponseEntity.badRequest().body(ExceptionResponse.builder().message(exception.getMessage()).errorCode(400).description(exception.toString()).build());
```

Comment:
you can just put `NotFoundException` in the body, there is no need to create another class and for below

Reply:
Done.